File: day3.py
'''c=int(input())
a=bin(c)[2:]
count=0
for i in a:
    if i=='1':
        count=count+1
print(count)'''
"""def isPowerOfTwo(num):
    if num <= 0:
        return False  # Non-positive numbers are not powers of 2.
    return (num & (num - 1)) == 0

# Test the function
num = int(input("Enter a number: "))
if isPowerOfTwo(num):
    print(f"{num} is a power of 2.")
else:
    print(f"{num} is not a power of 2.")
"""
'''import math

def isPowerOfFour(num):
    if num <= 0:
        return False  # Non-positive numbers are not powers of 4.

    log_value = math.log(num, 4)
    return log_value.is_integer()

# Test the function
num = int(input("Enter a number: "))
if isPowerOfFour(num):
    print(f"{num} is a power of 4.")
else:
    print(f"{num} is not a power of 4.")'''
''' power of 4
def isPowerOfTwo(num):
    if num <= 0:
        return False  # Non-positive numbers are not powers of 2.
    return (num & (num - 3)) == 0

# Test the function
num = int(input("Enter a number: "))
if isPowerOfTwo(num):
    print(f"{num} is a power of 4.")
else:
    print(f"{num} is not a power of 4.")'''
'''
MISSING NO
def find_missing_number(arr):
    n = len(arr) + 1  # Calculate the expected length of the array
    expected_sum = (n * (n + 1)) // 2  # Calculate the expected sum of consecutive integers
    
    actual_sum = sum(arr)  # Calculate the actual sum of the elements in the array
    
    missing_number = expected_sum - actual_sum  # The difference is the missing number
    
    return missing_number

# Example usage:
arr = [1, 2, 4, 5, 6]
missing_number = find_missing_number(arr)
print(f"The missing number is: {missing_number}")'''
'''def find_lonely_number(arr):
    lonely_number = 0
    
    for num in arr:
        lonely_number ^= num
    
    return lonely_number

# Example usage:
arr = [1, 2, 3, 4, 3, 2, 1]
lonely_number = find_lonely_number(arr)
print(f"The lonely number is: {lonely_number}")'''
'''import math
'''
factorial_100000 = math.factorial(100000)
print(factorial_100000)'''
def toh(n,source,auxilary,destination):
    if n==0:
        return
    toh(n-1,source,destination,auxilary)
    print("disk",n,"moved from",source,"to",dest)
    toh(n-1,aux,source,dest)
n=int(input())
source,aux,dest=input().split()
toh(n,source,aux,dest)
o/p:
    3
a b c
disk 1 moved from a to c
disk 2 moved from a to c
disk 1 moved from b to c
disk 3 moved from a to c
disk 1 moved from b to c
disk 2 moved from b to c
disk 1 moved from b to c'''

File: day3.c
/*main()
{

int a=12;
if(a&1)
{
	printf("odd");
}
else
{
	printf("even");//even
}
}*/
/*main()
{
	int a=15,count=0;
	while(a)
	{
		count++;
		a=a&(a-1);
	}
	printf("count=%d",count);//4
}*/
/*smallest no greater than n with exactly 1 bit diff in binary form
main()
{
	int a=14;
	printf("%d",a|a+1);
}*/
/*main()
{
	int a=15,i=4;
	if(a&(1<<(i-1)))
	{
		printf("yes");//yes
	}
	else
	{
		printf("no");
	}
}*/
//determine a number is power of 2 or not 
/*#include <stdio.h>

int isPowerOfTwo(int num) {
    if (num <= 0) {
        return 0; 
    }
    return (num & (num - 1)) == 0;
}

int main() {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);

    if (isPowerOfTwo(num)) {
        printf("%d is a power of 2.\n", num);
    } else {
        printf("%d is not a power of 2.\n", num);
    }

    return 0;
}*/
//determine a number is power of 4 or not 
/*#include <stdio.h>
#include <math.h>

int isPowerOfFour(int num) {
    if (num <= 0) {
        return 0; 
    }

    double logValue = log(num) / log(4);
    return (logValue - (int)logValue) == 0;
}

int main() {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);

    if (isPowerOfFour(num)) {
        printf("%d is a power of 4.\n", num);
    } else {
        printf("%d is not a power of 4.\n", num);
    }

    return 0;
}*/
// power of 2
/*main()
{

int count=0,n=256;
while(n)
{
	count++;
	n=n&(n-1);
}
if(count==1)
printf("true");//true
else
printf("false");
}*/
// power of 4
/*main()
{

int count=0,n=256;
while(n)
{
	count++;
	n=n&(n-3);
}
if(count==1)
printf("true");//true
else
printf("false");
}*/
/*main()
{
if(printf("%d",1))
printf("test");//1test
else
printf("no");
}*/
// 
// missing no
/*main()
{
int a[]={1,2,3,4,5,6,7,9};
int x=0,i=0;
for(i=0;i<=8;i++)
x=x^i;
for(i=0;i<8-1;i++)
x=x^a[i];
printf("%d",x);//8
}*/
/*main()
{
int a[5]={10,20,30,20,10};
int res=0,i=0;
for(i=0;i<5;i++)
{
	res=res^a[i];
}
printf("%d",res);//30
}*/
/*main()
{
int a[]={10,20,30,20,10,35,10};
int res=0,i=0;
for(i=0;i<=6;i++)
{
	res=res^a[i];
//30
}
printf("%d",res);
}*/
// write a factorial no using recursion
/*#include<stdio.h>
int main()
{
	long long int n,i,fact=1;
	scanf("%lld",&n);
	for(i=1;i<=n;i++)
	{
		fact=fact*i;
	}
	printf("%lld",fact);
}
*/
/*#include<stdio.h>
long long int fib(long long int n)
{
	if(n==0||n==1)
	return n;
	else
	return fib(n-1)+fib(n-2);
}
int main() 
{
	long long int n,i;
	scanf("%lld",&n);
	for(i=0;i<n;i++)
	printf("%lld\n",fib(i));
}*/
main()
{
	int n,i;
	
}